vnext/Chakra/ChakraExecutor.cpp

```diff
@@ -512,17 +504,20 @@ void ChakraExecutor::loadApplicationScript(
 
   ChakraString jsSourceURL(sourceURL.c_str());
 
+#if defined(OSS_RN)
+  evaluateScript(std::move(script), jsSourceURL);
+#else
+  const ScriptMetadata *scriptMetadata = nullptr;
+  if (m_instanceArgs.ScriptMetadata) {
+    scriptMetadata = &m_instanceArgs.ScriptMetadata->at(sourceURL);
```

Comment:
>->at [](start = 51, length = 4)

What happens if it doesn't exist? exception?

Reply:
Yep. We'll throw an exception. Might be better to just do a find() and do nothing if we get the end iter, since we already have a "no bytecode available" path

---
In reply to: [342747775](https://github.com/microsoft/react-native-windows/pull/3591#discussion_r342747775) [](ancestors = 342747775)

Reply:
The lines for what can be null, non-null, filled, not-filled in DevSettings is getting pretty blurry. I will document this one a bit better.

---
In reply to: [342859393](https://github.com/microsoft/react-native-windows/pull/3591#discussion_r342859393) [](ancestors = 342859393,342747775)

Reply:
Now that we have C++17, should we use std::optional here to express optional parameters in DevSettings?

---
In reply to: [342859664](https://github.com/microsoft/react-native-windows/pull/3591#discussion_r342859664) [](ancestors = 342859664,342859393,342747775)

Reply:
I'm personally not a fan of optional for types that are already nullable like shared_ptr. For those, I think documentation and maybe assertion during init that everything non-optional could clear things up. I'll add a comment for what I just added and log an item to look at cleaning things up.

---
In reply to: [342861473](https://github.com/microsoft/react-native-windows/pull/3591#discussion_r342861473) [](ancestors = 342861473,342859664,342859393,342747775)